lib/errors/invalid-argument-error.js

```diff
@@ -0,0 +1,33 @@
+
+/**
+ * Module dependencies.
+ */
+
+var _ = require('lodash');
+var StandardHttpError = require('standard-http-error');
```

Comment:
[Any error that does not inherit](https://github.com/thomseddon/node-oauth2-server/pull/203/files#diff-981f77ec338907079df6969aeb490d70R92) from "OAuthError" will be wrapped into a "ServerError".

The idea is that, if an unhandled error is thrown, it'll bubble up to the application where it would (presumably) be converted into an HTTP 500 error by the web framework.

At the same time, we want to allow users of the _node-oauth2-server_ package to throw an _OAuthError_ (for example: an InvalidScopeError) at any time, which will be correctly converted into the proper OAuth response.

In this case, there are two classes that inherit from "standard-http-error":

* _OAuthError_: this is the base class for all "operational" errors, as explained above.
* _InvalidArgumentError_: this is a programmatic error intended to help our users determine what went wrong with their server implementation. This will not be wrapped into an OAuth response and will bubble up as an HTTP 500 error (and rightly so, in my opinion). _Note that this could just as easily be a simple `throw new Error()` but I prefer using classes for errors since they allow wrappers to easily determine the type of error._